R/modules.R

```diff
@@ -1,179 +1,24 @@
-#'               output$data <- renderDataTable(data()[["iris"]])
-#'             }
-#'           )
-#'         },
-#'         ui = function(id) {
-#'           ns <- NS(id)
-#'           tagList(dataTableOutput(ns("data")))
-#'         },
-#'         datanames = "all"
-#'       )
-#'     ),
-#'     module(
-#'       label = "Another module",
-#'       server = function(id) {
-#'         moduleServer(
-#'           id,
-#'           module = function(input, output, session) {
-#'             output$text <- renderText("Another module")
-#'           }
-#'         )
-#'       },
-#'       ui = function(id) {
-#'         ns <- NS(id)
-#'         tagList(textOutput(ns("text")))
-#'       },
-#'       datanames = NULL
-#'     )
-#'   )
-#' )
-#' if (interactive()) {
-#'   shinyApp(app$ui, app$server)
-#' }
-modules <- function(..., label = "root") {
-  checkmate::assert_string(label)
-  submodules <- list(...)
-  if (any(vapply(submodules, is.character, FUN.VALUE = logical(1)))) {
-    stop(
-      "The only character argument to modules() must be 'label' and it must be named, ",
-      "change modules('lab', ...) to modules(label = 'lab', ...)"
-    )
-  }
-
-  checkmate::assert_list(submodules, min.len = 1, any.missing = FALSE, types = c("teal_module", "teal_modules"))
-  # name them so we can more easily access the children
-  # beware however that the label of the submodules should not be changed as it must be kept synced
-  labels <- vapply(submodules, function(submodule) submodule$label, character(1))
-  names(submodules) <- make.unique(gsub("[^[:alnum:]]+", "_", labels), sep = "_")
-  structure(
-    list(
-      label = label,
-      children = submodules
-    ),
-    class = "teal_modules"
-  )
-}
-
-#' Append a `teal_module` to `children` of a `teal_modules` object
-#' @keywords internal
-#' @param modules (`teal_modules`)
-#' @param module (`teal_module`) object to be appended onto the children of `modules`
-#' @return A `teal_modules` object with `module` appended.
-append_module <- function(modules, module) {
-  checkmate::assert_class(modules, "teal_modules")
-  checkmate::assert_class(module, "teal_module")
-  modules$children <- c(modules$children, list(module))
-  labels <- vapply(modules$children, function(submodule) submodule$label, character(1))
-  names(modules$children) <- make.unique(gsub("[^[:alnum:]]", "_", tolower(labels)), sep = "_")
-  modules
-}
-
-#' Extract/Remove module(s) of specific class
+#' @param label (`character(1)`) Label shown in the navigation item for the module or module group.
+#'   For `modules()` defaults to `"root"`.
+#'   Note that for `modules()` `label` comes after `...`, so it must be passed as a named argument,
+#'   otherwise it will be captured by `...`.
 #'
-#' Given a `teal_module` or a `teal_modules`, return the elements of the structure according to `class`.
+#'   `"global_filters"` is reserved - see `mapping` argument of [teal_slices()].
```

Comment:
What if we move this part

> "global_filters" is reserved - see mapping argument of [teal_slices()]
> "Report previewer" is reserved - see [reporter_previewer_module()]


to details? The info between label and server is getting stuffed/jam-packed in the help page

<img width="538" alt="image" src="https://github.com/insightsengineering/teal/assets/133694481/26fdb4c2-2382-480e-9832-9dc8800520ab">

Reply:
Good point, done.